tripal_chado/includes/TripalFields/obi__organism/obi__organism.inc

```diff
@@ -124,19 +124,23 @@ class obi__organism extends ChadoField {
         return;
       }
       $string = $settings['field_display_string'];
+      // The infraspecific fields were introduced in Chado v1.3, include only if present.
+      // Special case conversion of the type_id 'no_rank' value to an empty string.
+      if (property_exists($organism, 'type_id')) {
+        if ($organism->type_id->name == 'no_rank') {
+          $organism->type_id->name = '';
```

Comment:
Rather than change the `$organism->type_id->name` value how about just create a variable in this function for the infraspecific name that can get set and used below.  I'm not 100% sure if changing the value of the object means that when it is used in other fields if that change is persistent.

Reply:
That's tricky, because right after that it is used in
```$label = chado_replace_tokens($string, $organism);```
to construct the label, which is where the visible displayed text is created.

Reply:
Lol. that was my original idea, but it seemed "wrong". It is a lot cleaner and simpler, though. I will push my extensive changes for the record, and then likely do that.